apiserver/facades/client/highavailability/highavailability.go

```diff
@@ -197,7 +197,7 @@ func validateCurrentControllers(st *state.State, cfg controller.Config, machineI
 			// machines without any address are essentially not started yet
 			continue
 		}
-		internal := network.SelectInternalAddresses(addresses, false)
+		internal := addresses.AllMatchingScope(network.ScopeMatchCloudLocal)
```

Comment:
Example of the generalised scope-matching methods here. The caller has the burden of selecting a matcher.

I think that this technique of enriching the collection type with a method instead of have a public method in package makes for a better developer experience too.